cocotb/share/lib/vhpi/VhpiImpl.cpp

```diff
@@ -987,6 +977,34 @@ void VhpiImpl::sim_end() {
     }
 }
 
+bool VhpiImpl::compare_names(const std::string &a, const std::string &b) {
+#ifdef NVC
+    /* NVC does not properly implement the CaseName property and returns
+       Names instead (nickg/nvc#723). */
+    return a.size() == b.size() &&
+           equal(a.begin(), a.end(), b.begin(), [](char x, char y) {
+               return std::toupper(x) == std::toupper(y);
+           });
+#else
+    return a == b;
+#endif
+}
+
+std::string VhpiImpl::fully_qualified_name(const std::string &name,
```

Comment:
These two added functions should be functionally equivalent to the existing code when `NVC` is not defined. I can extract them into a separate PR without the `#ifdef NVC` bits if that's easier to review.